src/main/groovy/org/edx/jenkins/dsl/rest/RestApiJobManagement.groovy

```diff
@@ -0,0 +1,121 @@
+package org.edx.jenkins.dsl.rest
+
+import groovyx.net.http.ContentType
+import groovyx.net.http.HttpResponseDecorator
+import groovyx.net.http.RESTClient
+import javaposse.jobdsl.dsl.*
+
+class RestApiJobManagement extends MockJobManagement {
+
+    final RESTClient restClient
+
+    RestApiJobManagement(String baseUrl) {
+        restClient = new RESTClient(baseUrl)
+        restClient.handler.failure = { it }
+    }
+
+    void setCredentials(String username, String password) {
+        restClient.headers['Authorization'] = 'Basic ' + "$username:$password".bytes.encodeBase64()
+    }
+
+    @Override
+    String getConfig(String jobName) throws JobConfigurationNotFoundException {
+        String xml = fetchExistingXml(jobName)
+        if (!xml) {
+            throw new JobConfigurationNotFoundException(jobName)
+        }
+
+        xml
+    }
+
+    @Override
+    boolean createOrUpdateConfig(Item item, boolean ignoreExisting) throws NameNotProvidedException {
+        createOrUpdateConfig(item.name, item.xml, ignoreExisting, false)
+    }
+
+    @Override
+    void createOrUpdateView(String viewName, String config, boolean ignoreExisting) throws NameNotProvidedException, ConfigurationMissingException {
+        createOrUpdateConfig(viewName, config, ignoreExisting, true)
+    }
+
+    boolean createOrUpdateConfig(String name, String xml, boolean ignoreExisting, boolean isView) throws NameNotProvidedException {
+        boolean success
+        String status
+
+        String existingXml = fetchExistingXml(name, isView)
+        if (existingXml) {
+            if (ignoreExisting) {
+                success = true
+                status = 'ignored'
+            } else {
+                success = update(name, xml, isView)
+                status = success ? 'updated' : 'update failed'
+            }
+        } else {
+            success = create(name, xml, isView)
+            status = success ? 'created' : 'create failed'
+        }
+
+        println "$name - $status"
+        success
+    }
+
+    @Override
+    InputStream streamFileInWorkspace(String filePath) throws IOException {
+        new File(filePath).newInputStream()
+    }
+
+    @Override
+    String readFileInWorkspace(String filePath) throws IOException {
+        new File(filePath).text
+    }
+
+    private boolean create(String name, String xml, boolean isView) {
```

Comment:
This is all good stuff but I admit I'm wondering if it's in a library that we could just import instead?